src/goto-instrument/synthesizer/cegis_verifier.cpp

```diff
@@ -0,0 +1,482 @@
+          // is a loop havocing instruction.
+          if(is_loop_havoc(step.pc))
+          {
+            havoced_values[step.full_lhs] =
+              hex_to_size_t(step.full_lhs_value.get_string(ID_value));
+          }
+        }
+
+        // Record object_size, pointer_offset, and loop_entry(pointer_offset).
+        if(
+          can_cast_expr<annotated_pointer_constant_exprt>(
+            step.full_lhs_value) &&
+          contains_symbol_prefix(
+            step.full_lhs_value, SYMEX_DYNAMIC_PREFIX "dynamic_object"))
+        {
+          const auto &annotated_pointer_constant_expr =
+            expr_try_dynamic_cast<annotated_pointer_constant_exprt>(
+              step.full_lhs_value);
+
+          object_sizes[step.full_lhs] =
+            extract_size(annotated_pointer_constant_expr->symbolic_pointer());
+
+          // Store the offset into the map for loop_entry if we haven't
+          // entered the loop.
+          if(!entered_loop)
+          {
+            loop_entry_offsets[step.full_lhs] = extract_offset(
+              annotated_pointer_constant_expr->symbolic_pointer());
+          }
+
+          // Store the offset into the the map for havoced offset if this step
+          // is a loop havocing instruction.
+          if(is_loop_havoc(step.pc))
+          {
+            havoced_pointer_offsets[step.full_lhs] = extract_offset(
+              annotated_pointer_constant_expr->symbolic_pointer());
+          }
+        }
+      }
+    }
+
+    case goto_trace_stept::typet::ASSERT:
+    case goto_trace_stept::typet::FUNCTION_CALL:
+    case goto_trace_stept::typet::FUNCTION_RETURN:
+    case goto_trace_stept::typet::ASSUME:
+    case goto_trace_stept::typet::LOCATION:
+    case goto_trace_stept::typet::GOTO:
+    case goto_trace_stept::typet::OUTPUT:
+    case goto_trace_stept::typet::INPUT:
+    case goto_trace_stept::typet::SPAWN:
+    case goto_trace_stept::typet::MEMORY_BARRIER:
+    case goto_trace_stept::typet::ATOMIC_BEGIN:
+    case goto_trace_stept::typet::ATOMIC_END:
+    case goto_trace_stept::typet::DEAD:
+    case goto_trace_stept::typet::CONSTRAINT:
+    case goto_trace_stept::typet::SHARED_READ:
+    case goto_trace_stept::typet::SHARED_WRITE:
+      break;
+
+    case goto_trace_stept::typet::NONE:
+      UNREACHABLE;
+    }
+  }
+
+  return cext(
+    object_sizes,
+    havoced_values,
+    havoced_pointer_offsets,
+    loop_entry_values,
+    loop_entry_offsets,
+    live_variables);
+}
+
+void cegis_verifiert::restore_functions()
+{
+  for(const auto &fun_entry : goto_model.goto_functions.function_map)
+  {
+    irep_idt fun_name = fun_entry.first;
+    goto_model.goto_functions.function_map[fun_name].body.swap(
+      original_functions[fun_name]);
+  }
+}
+
+bool cegis_verifiert::verify()
+{
+#if DEBUG
+#else
+  null_message_handlert null_message_handler;
+  ui_message_handlert ui_null_message_handler(null_message_handler);
+  messaget null_log(ui_null_message_handler);
+  log = null_log;
+#endif
```

Comment:
This looks suspicious to me. We have `.debug()`, which should be debug-only output.